beluga_tutorial/scripts/visualize.py

```diff
@@ -0,0 +1,162 @@
+
+"""Visualization Scripts for Beluga Tutorial.
+
+This script uses the matplotlib to visualize the data generated
+by beluga_tutorial example code.
+"""
+
+import yaml
+import matplotlib.pyplot as plt
+import matplotlib.animation as animation
+import numpy as np
+import argparse
+
+
+def plot_stages(yaml_data, axs, index):
+    particles = yaml_data['simulation_records'][index]["particles"]
+    for j, stage in enumerate(['current', 'propagate', 'reweight', 'resample']):
+        states = particles[stage]['states']
+        ax = axs[j]
+        ax.clear()
+        ax.hist(
+            states,
+            bins=np.arange(min(states), max(states) + 0.5, 0.5),
+            color='skyblue',
+            edgecolor='black',
+        )
+        ax.set_title(f"{stage} - Sim Cycle {index+1}")
+        ax.set_xlabel("States")
+        ax.set_ylabel("NP")
+        ax.set_xlim(-5, 100)
+        ax.set_xticks(np.arange(-5, 101, 5))
+
+    landmark_map = yaml_data["landmark_map"]
+    ax_landmark = axs[4]
+    ax_landmark.clear()
+
+    for x in range(101):
+        if x in landmark_map:
+            color = 'red'
+        else:
+            color = 'blue'
+        ax_landmark.bar(x, 1, color=color)
+
+    ax_landmark.set_title("Landmark Map")
+    ax_landmark.set_xlabel("States")
+    ax_landmark.set_xlim(-5, 100)
+    ax_landmark.set_ylim(0, 2)
+    ax_landmark.set_xticks(np.arange(-5, 101, 5))
+
+    ground_truth = yaml_data['simulation_records'][index]["ground_truth"]
+    ax_ground_truth = axs[5]
+    ax_ground_truth.clear()
+    ax_ground_truth.bar(ground_truth, 1, color='green')
+    ax_ground_truth.set_title(f"Ground Truth: {ground_truth}")
+    ax_ground_truth.set_xlabel("States")
+    ax_ground_truth.set_xlim(-5, 100)
+    ax_ground_truth.set_ylim(0, 2)
+    ax_ground_truth.set_xticks(np.arange(-5, 101, 5))
+
+    mean = "{:.3f}".format(yaml_data['simulation_records'][index]["estimation"]["mean"])
+    sd = "{:.3f}".format(yaml_data['simulation_records'][index]["estimation"]["sd"])
+    plt.text(
+        0.5,
+        0.5,
+        f"Mean: {mean}\nSD: {sd}",
+        ha='center',
+        va='center',
+        transform=axs[5].transAxes,
+        bbox=dict(facecolor='white', alpha=0.5),
+    )
+
+    plt.tight_layout()
+    plt.draw()
+
+
+def on_key(event, yaml_data, axs, num_frames):
+    if not hasattr(on_key, "current_frame"):
+        on_key.current_frame = 0
+
+    if event.key == 'right':
+        on_key.current_frame = (on_key.current_frame + 1) % num_frames
+    elif event.key == 'left':
+        on_key.current_frame = (on_key.current_frame - 1) % num_frames
+
+    plot_stages(yaml_data, axs, on_key.current_frame)
+
+
+def main(argv=None) -> int:
+    parser = argparse.ArgumentParser(description=globals()['__doc__'])
+
+    parser.add_argument(
+        '-p',
+        '--record-file-path',
+        type=str,
+        help='Absolute path to the record file generated by the beluga_tutorial example code',
+        required=True,
+    )
```

Comment:
Consider making this a positional argument, so we can pass the path directly without `-p`.